base/parse.jl

```diff
@@ -171,11 +171,45 @@ end
     throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 62, got $base"))
 end
 
+
+"""
+    tryparse(type, str, [base])
+
+Like [`parse`](@ref), but returns a [`Nullable`](@ref) of the requested type. The result will be null if the
+string does not contain a valid number.
+"""
+function tryparse end
+
 tryparse(::Type{T}, s::AbstractString, base::Integer) where {T<:Integer} =
     tryparse_internal(T, s, start(s), endof(s), check_valid_base(base), false)
 tryparse(::Type{T}, s::AbstractString) where {T<:Integer} =
     tryparse_internal(T, s, start(s), endof(s), 0, false)
 
+
+"""
+    parse(type, str, [base])
+
+Parse a string as a number. If the type is an integer type, then a base can be specified
+(the default is 10). If the type is a floating point type, the string is parsed as a decimal
+floating point number. If the string does not contain a valid number, an error is raised.
+
+```jldoctest
+julia> parse(Int, "1234")
+1234
+
+julia> parse(Int, "1234", 5)
+194
+
+julia> parse(Int, "afc", 16)
+2812
+
+julia> parse(Float64, "1.2e-3")
+0.0012
+```
+"""
+function parse(T::Type, str, base=Int) end
```

Comment:
Looks like it was a manual mistake in https://github.com/JuliaLang/julia/commit/9e6aecc1b34418cac82d667fdf9f26b1eac596c4 that no one noticed

In the 3-arg method with a base it's constrained to be `::Integer`. In the 2-arg method below 0 gets passed instead of `check_valid_base(base)`, but a comment says `# Zero means, "figure it out"` which isn't consistent with "the default is 10" from the docstring. `parse(Int, "0b01", 10)` errors and `parse(Int, "0b01")` doesn't...